src/main/java/com/startstepszalando/ecommerceshop/user/controller/UserController.java

```diff
@@ -21,31 +35,80 @@
-    ) throws DuplicateUserException {
-        return ResponseEntity.ok(userService.registerUser(request));
+    ) {
+        try {
+            AuthenticationResponse response = userService.registerUser(request);
+            return ResponseEntity.ok(response);
+        } catch (DuplicateUserException e) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body("Duplicate user with the same email already exists");
+        }
+    }
+
+    @Operation(summary = "User login", responses = {
+            @ApiResponse(responseCode = "200", description = "User logged in successfully",
+                    content = @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = AuthenticationResponse.class))),
+            @ApiResponse(responseCode = "400", description = "Invalid user details or validation error",
+                    content = @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = InvalidUserDetailsException.class))),
+            @ApiResponse(responseCode = "401", description = "Invalid credentials",
+                    content = @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = InvalidCredentialsException.class))),
+            @ApiResponse(responseCode = "404", description = "User not found",
+                    content = @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = UserNotFoundException.class))),
+            @ApiResponse(responseCode = "500", description = "Internal server error",
+                    content = @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = Exception.class)))
+    })
+    @PostMapping(value = "/login")
+    public ResponseEntity<?> login(@Valid @RequestBody UserLoginRequest loginRequest) {
+
+        if (loginRequest.getEmail() == null || loginRequest.getEmail().trim().isEmpty() || !loginRequest.getEmail().contains("@")) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body("Please provide a valid email address");
+        }
+
+        try {
+            Authentication authentication = authenticationManager.authenticate(
+                    new UsernamePasswordAuthenticationToken(loginRequest.getEmail(), loginRequest.getPassword())
+            );
+
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+            UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+            final String jwt = jwtService.generateToken(userDetails);
+
+            return ResponseEntity.ok(new AuthenticationResponse(jwt, "User logged in successfully"));
+        } catch (AuthenticationException e) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED)
+                    .body("Invalid username and/or password");
```

Comment:
I'm having an issue getting the `GlobalExceptionHandler` being read for 400/401 errors, and need help with this.